src/lib.rs

```diff
@@ -0,0 +1,1040 @@
+}
+
+impl HasSecurityContext for apicore::EphemeralContainer {
+    fn get_security_context(&self) -> Option<apicore::SecurityContext> {
+        self.security_context.clone()
+    }
+}
+
+/* Validates the pod annotations. The seccomp annotations should be defined in
+ * the settings.
+ */
+fn do_validate_annotations(pod: &apicore::Pod, settings: &settings::Settings) -> Result<()> {
+    if let Some(annotations) = &pod.metadata.annotations {
+        let mut invalid_profiles = vec![];
+        for (annotation, profile) in annotations {
+            if (annotation.starts_with("container.seccomp.security.alpha.kubernetes.io")
+                || annotation == "seccomp.security.alpha.kubernetes.io/pod")
+                && !settings.allowed_profiles.contains(profile)
+            {
+                invalid_profiles.push(profile.clone())
+            }
+        }
+        if !invalid_profiles.is_empty() {
+            return Err(anyhow!(
+                "Seccomp profiles '{}' are not allowed.",
+                invalid_profiles.join(",")
+            ));
+        }
+    }
+    Ok(())
+}
+
+/* Validates the Seccomp configuration of the given containers.
+ * Checks the type values and if the localhost profile is set when necessary
+ */
+fn do_validate_containers<C>(containers: &[C], settings: &settings::Settings) -> Result<()>
+where
+    C: HasSecurityContext,
+{
+    let mut invalid_seccomp_profile_types = vec![];
+    let mut invalid_seccomp_profile = vec![];
+    for container in containers {
+        if let Some(security_context) = container.get_security_context() {
+            if let Some(seccomp_profile) = security_context.seccomp_profile.clone() {
+                if !settings.profile_types.contains(&seccomp_profile.type_) {
+                    invalid_seccomp_profile_types.push(seccomp_profile.type_.clone())
+                }
+                if seccomp_profile.type_ == "Localhost" {
+                    if let Some(localhost_profile) = &seccomp_profile.localhost_profile {
+                        if !settings.localhost_profiles.contains(localhost_profile) {
+                            invalid_seccomp_profile.push(localhost_profile.clone())
+                        }
+                    } else {
+                        return Err(anyhow!(
+                            "The localhost seccomp profile must be set.".to_string(),
+                        ));
+                    }
+                }
+            }
+        }
+    }
+    if !invalid_seccomp_profile_types.is_empty() {
+        return Err(anyhow!(format!(
+            "Invalid seccomp profile types: {}",
+            invalid_seccomp_profile_types.join(",")
+        )));
+    }
+    if !invalid_seccomp_profile.is_empty() {
+        return Err(anyhow!(format!(
+            "Invalid seccomp profile: {}",
+            invalid_seccomp_profile.join(",")
+        )));
+    }
+    Ok(())
+}
+
+fn do_validate_pod(pod: &apicore::PodSpec, settings: &settings::Settings) -> Result<()> {
```

Comment:
nitpicking, the name gives me the impression this is going to validate all the possible contents of a Pod (which includes also its containers, init containers and ephemeral containers).

I would rename this function to be `do_validate_pod_security_context`